src/clj/game/cards/events.clj

```diff
@@ -1051,6 +1051,19 @@
     :choices (req runnable-servers)
     :effect (effect (run target nil card))}
 
+   "Insight"
+   {:async true
+    :implementation "Rearranging R&D is not optional"
+    :effect (req
+             (let [from (take 4 (:deck corp))]
+               (if (pos? (count from))
+                 (do (show-wait-prompt state :runner "Corp to rearrange the top cards of R&D")
```

Comment:
Might want to say how many cards Corp is rearranging, so "top `(count from)` cards".